Applications/Spire/Include/Spire/Ui/ArrayTableModel.hpp

```diff
@@ -71,10 +72,25 @@ namespace Spire {
 
       boost::signals2::connection connect_operation_signal(
         const typename OperationSignal::slot_type& slot) const override;
+
+    private:
+      mutable OperationSignal m_operation_signal;
+      std::vector<std::vector<std::any>> m_data;
+      Transaction m_transaction;
+      int m_transaction_level;
   };
 
   template<typename F>
   decltype(auto) ArrayTableModel::transact(F&& transaction) {
+    m_transaction_level++;
+    BOOST_SCOPE_EXIT(this_) {
+      this_->m_transaction_level--;
+      if(this_->m_transaction_level == 0) {
+        this_->m_operation_signal(this_->m_transaction);
```

Comment:
If ```m_transaction.m_operations.size() == 1```, then signal it directly instead of signaling a transaction. Add a unit test for that scenario.